index.js

```diff
@@ -74,7 +75,22 @@ const onChange = (object, onChange, options = {}) => {
 			basePath = cache.getPath(target);
 		}
 
-		return cache.getProxy(value, path.concat(basePath, property), handler, proxyTarget);
+		// Check circular references
+		// If the value already has a corresponding path/proxy,
+		// And if the path corresponds to one of the parent of the current value
+		// Then we are on a circular case, where the child is pointing to his parent.
+		// => in this case we return the proxy object with the shortest path.
+		// Otherwise we can fell into an infinite loop,
+		// and the path can get longer and longer until we reach a memory limit.
+		const childPath = path.concat(basePath, property);
+		const existingPath = cache.getPath(value);
+		if (existingPath && !isArray(childPath) && !isSymbol(childPath) && childPath.toString().startsWith(existingPath.toString())) {
```

Comment:
Using `startsWith` is a bit fragile. It could potentially match the wrong thing. For example, `existingPath` is `foo` while `childPath` is `foo2.bar`.

Reply:
Yes, you're totaly right, I didn't thought about this. I've rewriten the code to do a better comparison.
Actually I didn't manage the arrays of paths correctly in my first pull-request. I changed the code to manage arrays as well. And I've added another unittest for that too.